etc/systemd/system-generators/zfs-mount-generator.c

```diff
@@ -0,0 +1,1060 @@
+/*
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+ * AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT
+ * OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+
+#include <sys/resource.h>
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <sys/mman.h>
+#include <semaphore.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <time.h>
+#include <regex.h>
+#include <search.h>
+#include <dirent.h>
+#include <string.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <errno.h>
+#include <libzfs.h>
+
+#define	STRCMP ((int(*)(const void *, const void *))&strcmp)
+#define	ABORT_ENOMEM() \
+	do { \
+		fprintf(stderr, PROGNAME ": %d: " \
+		    "not enough memory (L%d)!\n", getpid(), __LINE__); \
+		_exit(1 << 0); \
+	} while (0)
+
+#define	true ((_Bool)1)
```

Comment:
I'm not finding other uses of `_Bool` in the ZFS codebase. Everything seems to be `boolean_t`. And the `B_TRUE` and `B_FALSE`.

Reply:
That's an integer (well, enum), and I'm trying to have codegen that isn't horrific for no good reason.